### PR TITLE
remove bucharest gold, imagestream is broken

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@lance your changes to https://github.com/bucharest-gold/centos7-s2i-nodejs/blob/master/image-stream.yml broke this import so i'm removing it.  Please submit a PR for the new imagestream source if you want to keep this image in the library.

**Comment by lholmquist** [comment]
@bparees could you provide what exactly broke so we have a better idea of what to fix

**Comment by lance** [comment]
@bparees - some things

* I didn't know that this repository had a dependency on https://raw.githubusercontent.com/bucharest-gold/centos7-s2i-nodejs/master/image-streams.centos7.json
* Why would I submit a PR for something I'm completely unaware of and have no familiarity with? Can you please provide a little more information here. Asking around our team, we didn't know anything about this.
* The file name has been changed to image-stream.yml and can be found in the same location. Instead of doing this, you could have just changed the filename.

Cheers

File: official/jenkins/imagestreams/jenkins-rhel7.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
                 },
                 "from": {
                     "kind": "DockerImage",
-                    "name": "registry.access.redhat.com/openshift3/jenkins-2-rhel7:v3.9"
+                    "name": "registry.access.redhat.com/openshift3/jenkins-2-rhel7:v3.10"
```
Comment (lance) [inline official/jenkins/imagestreams/jenkins-rhel7.json]: What does this have to do with Node.js?

**Comment by lance** [comment]
Lots of changes in this PR to things related to django and rails. I don't think that's related to our stuff is it?

File: community/django/templates/django-psql-example.json [hunk 2]
```diff
@@ -341,11 +341,14 @@
                                 ],
                                 "image": " ",
                                 "livenessProbe": {
-                                    "initialDelaySeconds": 30,
-                                    "tcpSocket": {
-                                        "port": 5432
+                                    "exec": {
```
Comment (lance) [inline community/django/templates/django-psql-example.json]: How is this related to Node.js?

File: community/django/templates/django-psql-persistent.json [hunk 3]
```diff
@@ -358,11 +358,14 @@
                                 ],
                                 "image": " ",
                                 "livenessProbe": {
-                                    "initialDelaySeconds": 30,
-                                    "tcpSocket": {
-                                        "port": 5432
+                                    "exec": {
```
Comment (lance) [inline community/django/templates/django-psql-persistent.json]: How is this related to Node.js?

File: community/jenkins/imagestreams/jenkins-centos7.json [hunk 4]
```diff
@@ -49,7 +49,7 @@
                 },
                 "from": {
                     "kind": "DockerImage",
-                    "name": "docker.io/openshift/jenkins-2-centos7:v3.9"
+                    "name": "docker.io/openshift/jenkins-2-centos7:v3.10"
```
Comment (lance) [inline community/jenkins/imagestreams/jenkins-centos7.json]: How is this related to Node.js?

File: community/rails/templates/rails-pgsql-persistent.json [hunk 5]
```diff
@@ -405,11 +405,14 @@
                                 ],
                                 "image": " ",
                                 "livenessProbe": {
-                                    "initialDelaySeconds": 30,
-                                    "tcpSocket": {
-                                        "port": 5432
+                                    "exec": {
```
Comment (lance) [inline community/rails/templates/rails-pgsql-persistent.json]: How is this related to Node.js?

File: community/rails/templates/rails-postgresql-example.json [hunk 6]
```diff
@@ -388,11 +388,14 @@
                                 ],
                                 "image": " ",
                                 "livenessProbe": {
-                                    "initialDelaySeconds": 30,
-                                    "tcpSocket": {
-                                        "port": 5432
+                                    "exec": {
```
Comment (lance) [inline community/rails/templates/rails-postgresql-example.json]: How is this related to Node.js?

File: official/django/templates/django-psql-example.json [hunk 7]
```diff
@@ -341,11 +341,14 @@
                                 ],
                                 "image": " ",
                                 "livenessProbe": {
-                                    "initialDelaySeconds": 30,
-                                    "tcpSocket": {
-                                        "port": 5432
+                                    "exec": {
```
Comment (lance) [inline official/django/templates/django-psql-example.json]: How is this related to Node.js?

File: official/django/templates/django-psql-persistent.json [hunk 8]
```diff
@@ -358,11 +358,14 @@
                                 ],
                                 "image": " ",
                                 "livenessProbe": {
-                                    "initialDelaySeconds": 30,
-                                    "tcpSocket": {
-                                        "port": 5432
+                                    "exec": {
+                                        "command": [
```
Comment (lance) [inline official/django/templates/django-psql-persistent.json]: How is this related to Node.js?

File: official/rails/templates/rails-pgsql-persistent.json [hunk 9]
```diff
@@ -405,11 +405,14 @@
                                 ],
                                 "image": " ",
                                 "livenessProbe": {
-                                    "initialDelaySeconds": 30,
-                                    "tcpSocket": {
-                                        "port": 5432
+                                    "exec": {
```
Comment (lance) [inline official/rails/templates/rails-pgsql-persistent.json]: How is this related to Node.js?

File: official/rails/templates/rails-postgresql-example.json [hunk 10]
```diff
@@ -388,11 +388,14 @@
                                 ],
                                 "image": " ",
                                 "livenessProbe": {
-                                    "initialDelaySeconds": 30,
-                                    "tcpSocket": {
-                                        "port": 5432
+                                    "exec": {
```
Comment (lance) [inline official/rails/templates/rails-postgresql-example.json]: How is this related to Node.js?

**Comment by coreydaley** [comment]
@lance 
The error occurred because the name of that file changed, you can see it here at the end of the logs:
https://ci.openshift.redhat.com/jenkins/job/update_openshift_library/469/console

I am tracking down who/how/when the imagestream source was added and will report back shortly.

**Comment by coreydaley** [comment]
It looks like @luciddreamz added it with this pull request: https://github.com/openshift/library/commit/bef3c637854c7d29c56dd6acae44fd560c74e761

So @lance you probably didn't know you were breaking anything.  It's up to the maintainers of the image streams to make sure that they don't break compatibility with the openshift library, or to update the library in preparation for the changes, but in this case you had no idea that your change would break the library process.  

If you would like it added back I can submit a PR that restores it (with the new file name/location), if not, then we can consider this issue closed.

Please let us know what you decide.

**Comment by lance** [comment]
@coreydaley thanks so much for the explanation. Over lunch I was starting to wonder if I'd added it in an insomnia-fueled late-night coding session and just forgot about it. 

In spite of the fact that I was unaware of this, I'm happy that we ~~are~~ were included! 

I don't mind submitting a PR, but honestly I am confused about all of the changes in this issue that are unrelated to Node.js. Did those just come along for the ride? If it's only a matter of reverting this commit and updating the filename, that seems pretty simple, but I don't want to make assumptions about all of those other changes.

I'll try to keep abreast of what's going on in this repo for now so that I can avoid this kind of mistake in the future.

**Comment by coreydaley** [comment]
@lance every night the library job runs and pulls in all of the changes from any template or imagestream listed in the community.yaml or official.yaml files, so it just happened that there were other changes also.

Submitting a pull request that reverts the commit that removed it, and updating the filename would work.  You will need to make sure to run the "make import" command and commit all of the changed files so that the tests will pass.
